system/Database/SQLite3/Connection.php

```diff
@@ -70,6 +79,10 @@ public function initialize()
         if (is_int($this->busyTimeout)) {
             $this->connID->busyTimeout($this->busyTimeout);
         }
+
+        if (is_int($this->synchronous)) {
```

Comment:
It might be a good idea to add validation or throw a database exception in this section. Since we are directly passing `$this->synchronous` into the PRAGMA statement, without validating whether the value is one of the expected integers `(0, 1, 2, 3)`, we risk unexpected behavior or incorrect PRAGMA settings if an invalid value is provided. Adding validation would ensure we are handling only valid inputs and would prevent potential silent failures.

Reply:
I added synchronous value validation and test.

I think using an enum here would mess with simplicity a bit - but if more people say we should go this way I can add it. Although I don't think we have a single enum in the project at the moment (I might have missed something).

Reply:
I don't think using an enum is necessary in this case. The reason enums weren't used in the framework was due to older PHP versions, but I believe we should use enums where possible.

In general, I'm a fan of enums as they improve code readability, but I don't see a strong need for them here.